cmd/monaco/deploy/deploy.go

```diff
@@ -55,13 +56,26 @@ func Deploy(fs afero.Fs, deploymentManifestPath string, specificEnvironments []s
 		return errors.New("error while loading manifest")
 	}
 
-	environments, err := manifest.FilterEnvironmentsByNames(specificEnvironments)
-	if err != nil {
-		return err
+	environments := manifest.Environments
+	if environmentGroup != "" {
+		environments = environments.FilterByGroup(environmentGroup)
+
+		if len(environments) == 0 {
+			return fmt.Errorf("no environments in group %q", environmentGroup)
+		} else {
+			log.Info("Environments loaded in group %q: %v", environmentGroup, maps.Keys(environments))
+		}
+
+	}
+
+	if len(specificEnvironments) > 0 {
+		environments, err = environments.FilterByNames(specificEnvironments)
+		if err != nil {
+			return fmt.Errorf("failed to filter environments: %w", err)
+		}
 	}
 
-	environmentMap := toEnvironmentMap(environments)
```

Comment:
seems these methods could now be removed?

Reply:
Yes, removed